.github/workflows/e2e-tests.yml

```diff
@@ -0,0 +1,62 @@
+name: End-to-end tests
+on:
+  push:
+    branches: ['main']
+
+jobs:
+  check-package-loads:
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+      - name: Setup poetry
+        uses: abatilo/actions-poetry@v2.0.0
+        with:
+          poetry-version: 1.1.13
+      - name: Cache poetry virtualenvs
+        uses: actions/cache@v3
+        with:
+          path: /home/runner/.cache/pypoetry/virtualenvs
+          key: poetry-venvs-no-dev-${{ runner.os }}-3.8-${{ hashFiles('poetry.lock') }}
+          restore-keys: poetry-vevns-no-dev-${{ runner.os }}-3.8-
+      - name: Check package loads without dev dependencies
+        run: make check-package-loads
+
+  e2e-tests:
+    needs: [ check-package-loads ]
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [ 3.7, 3.8 ]
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Setup poetry
+        uses: abatilo/actions-poetry@v2.0.0
+        with:
+          poetry-version: 1.1.13
+      - name: Cache poetry virtualenvs
+        uses: actions/cache@v3
+        with:
+          path: /home/runner/.cache/pypoetry/virtualenvs
+          key: poetry-venvs-${{ runner.os }}-${{ matrix.python-version }}-${{ hashFiles('dbt-adapters/poetry.lock') }}
+          restore-keys: poetry-venvs-${{ runner.os }}-${{ matrix.python-version }}-
+      - name: Install dbt
+        run: poetry add dbt
```

Comment:
How come this is needed in CI, but not in local? `dbt` should be in the poetry dev requreirements if not there already.

Reply:
My concern was that dbt can be installed in multiple ways (https://docs.getdbt.com/dbt-cli/install/overview), some of which I believe have implications like running the same dbt in multiple Python venvs. 
After some digging I discovered that they've actually deprecate the dbt package, so this logic definitely needs to get updated: https://docs.getdbt.com/dbt-cli/install/pip#pip-install-dbt

Reply:
Fair enough, if the solution is `poetry add` than that should just be in the dev dependencies. If it has to be in a different way then my comment is mute.

Reply:
This is resolved by calling dbt directly within python rather than in a subprocess